internal/app/app.go

```diff
@@ -60,14 +67,23 @@ func (a *App) Run(ctx context.Context) error {
 	// start messenger (receiver)
 	messenger, userIDUpdatedChan := a.startMessenger(gCtx, g)
 
-	// initiate tvm client
-	tvmClient, err := tvm.NewClient(a.cfg.TvmConfig) // TODO make client init conditional based on provided config
+	client, err := evm.NewClient(a.cfg.EvmConfig) // TODO make client init conditional based on provided config
 	if err != nil {
-		a.logger.Warn(err)
+		a.logger.Fatalf("Failed to connect to the Ethereum client: %v", err)
 	}
 
+	pk := new(secp256k1.PrivateKey)
+	// UnmarshalText expects the private key in quotes
+	if err := pk.UnmarshalText([]byte("\"" + a.cfg.PrivateKey + "\"")); err != nil {
+		a.logger.Fatalf("Failed to parse private key: %v", err)
+	}
+
+	// Get Ethereum Address from private key
+	cAddress := crypto.PubkeyToAddress(pk.ToECDSA().PublicKey)
```

Comment:
if we only use this PK as ECDSA pk, I'd suggest to directly pass it as ECDSA pk

Reply:
Because it is used only on C-Chain, it will be used only as ECDSA pk. Also, we should update the PK that is used to login to Matrix to use ECDSA as we are not using anything related to T-Chain now.

Reply:
> Because it is used only on C-Chain, it will be used only as ECDSA pk. Also, we should update the PK that is used to login to Matrix to use ECDSA as we are not using anything related to T-Chain now.

thats up for discussion, imo
I see both pros and cons

Reply:
_"we should already pass it as ecdsa key"_ - I did the change, I want to test it before resolving this last conversation

Reply:
Tested locally. Resolving this conversation -> ready for merge.